File: onboard_client.py
#!/usr/bin/env python3
"""
Onboard a new client by sending welcome email
Usage: python onboard_client.py <client_email> [--name CLIENT_NAME] [--calendar-link LINK]
"""

import os
import sys
import argparse
from pathlib import Path
from dotenv import load_dotenv

# Import send_email functions
from send_email import send_email_smtp, send_email_sendgrid

load_dotenv()


def load_template(template_name='welcome_email.html'):
    """Load email template from templates directory"""
    template_path = Path(__file__).parent / 'templates' / template_name

    if not template_path.exists():
        raise FileNotFoundError(f"Template not found: {template_path}")

    with open(template_path, 'r', encoding='utf-8') as f:
        return f.read()


def render_template(template_content, **variables):
    """Replace template variables with actual values"""
    rendered = template_content
    for key, value in variables.items():
        rendered = rendered.replace(f"{{{key}}}", value)
    return rendered


def onboard_client(client_email, client_name=None, calendar_link=None):
    """Send welcome email to new client"""

    # Get defaults from environment or use placeholders
    client_name = client_name or os.getenv('DEFAULT_CLIENT_NAME', 'there')
    calendar_link = calendar_link or os.getenv('CALENDAR_LINK', 'https://calendly.com/sevengravity')

    print(f"Onboarding client: {client_email}")
    print(f"Client name: {client_name}")
    print(f"Calendar link: {calendar_link}")

    # Load and render template
    try:
        template = load_template()
        email_body = render_template(
            template,
            client_name=client_name,
            calendar_link=calendar_link
        )
    except Exception as e:
        print(f"✗ Failed to load template: {e}")
        return False

    # Send email
    subject = "Welcome to Seven Gravity - Let's Get Started!"
    email_method = os.getenv('EMAIL_METHOD', 'smtp').lower()

    try:
        if email_method == 'sendgrid':
            success = send_email_sendgrid(client_email, subject, email_body)
        else:
            success = send_email_smtp(client_email, subject, email_body)

        if success:
            print(f"\n✓ Successfully onboarded {client_email}")
            print(f"✓ Welcome email sent with calendar link: {calendar_link}")
            return True
        else:
            print(f"\n✗ Failed to onboard {client_email}")
            return False

    except Exception as e:
        print(f"✗ Error sending email: {e}")
        return False


def main():
    parser = argparse.ArgumentParser(description='Onboard a new client')
    parser.add_argument('client_email', help='Client email address')
    parser.add_argument('--name', dest='client_name', help='Client name (optional)')
    parser.add_argument('--calendar-link', dest='calendar_link', help='Kickoff call calendar link (optional)')

    args = parser.parse_args()

    success = onboard_client(
        args.client_email,
        client_name=args.client_name,
        calendar_link=args.calendar_link
    )

    sys.exit(0 if success else 1)


if __name__ == '__main__':
    main()

File: onboard_client.js
#!/usr/bin/env node
/**
 * Onboard a new client by sending welcome email
 * Usage: node onboard_client.js <client_email> [--name CLIENT_NAME] [--calendar-link LINK]
 */

import nodemailer from 'nodemailer';
import { readFileSync } from 'fs';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';
import dotenv from 'dotenv';

// Load environment variables
dotenv.config();

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

function loadTemplate(templateName = 'welcome_email.html') {
  const templatePath = join(__dirname, 'templates', templateName);
  return readFileSync(templatePath, 'utf-8');
}

function renderTemplate(templateContent, variables) {
  let rendered = templateContent;
  for (const [key, value] of Object.entries(variables)) {
    rendered = rendered.replaceAll(`{${key}}`, value);
  }
  return rendered;
}

async function sendEmail(to, subject, htmlBody) {
  const transporter = nodemailer.createTransport({
    host: process.env.SMTP_HOST || 'smtp.gmail.com',
    port: parseInt(process.env.SMTP_PORT || '587'),
    secure: false,
    auth: {
      user: process.env.SMTP_USER,
      pass: process.env.SMTP_PASSWORD,
    },
  });

  const fromName = process.env.FROM_NAME || 'Seven Gravity';
  const fromEmail = process.env.FROM_EMAIL || process.env.SMTP_USER;

  try {
    const info = await transporter.sendMail({
      from: `"${fromName}" <${fromEmail}>`,
      to: to,
      subject: subject,
      html: htmlBody,
    });

    console.log(`✓ Email sent successfully to ${to}`);
    console.log(`✓ Message ID: ${info.messageId}`);
    return true;
  } catch (error) {
    console.error(`✗ Failed to send email: ${error.message}`);
    return false;
  }
}

async function onboardClient(clientEmail, clientName, calendarLink) {
  console.log(`Onboarding client: ${clientEmail}`);
  console.log(`Client name: ${clientName}`);
  console.log(`Calendar link: ${calendarLink}`);

  try {
    // Load and render template
    const template = loadTemplate();
    const emailBody = renderTemplate(template, {
      client_name: clientName,
      calendar_link: calendarLink,
    });

    // Send email
    const subject = "Welcome to Seven Gravity - Let's Get Started!";
    const success = await sendEmail(clientEmail, subject, emailBody);

    if (success) {
      console.log(`\n✓ Successfully onboarded ${clientEmail}`);
      console.log(`✓ Welcome email sent with calendar link: ${calendarLink}`);
    } else {
      console.log(`\n✗ Failed to onboard ${clientEmail}`);
    }

    return success;
  } catch (error) {
    console.error(`✗ Error: ${error.message}`);
    return false;
  }
}

// Parse command line arguments
const args = process.argv.slice(2);

if (args.length === 0) {
  console.log('Usage: node onboard_client.js <client_email> [--name CLIENT_NAME] [--calendar-link LINK]');
  process.exit(1);
}

const clientEmail = args[0];
let clientName = process.env.DEFAULT_CLIENT_NAME || 'there';
let calendarLink = process.env.CALENDAR_LINK || 'https://calendly.com/sevengravity';

// Parse optional arguments
for (let i = 1; i < args.length; i++) {
  if (args[i] === '--name' && args[i + 1]) {
    clientName = args[i + 1];
    i++;
  } else if (args[i] === '--calendar-link' && args[i + 1]) {
    calendarLink = args[i + 1];
    i++;
  }
}

// Run onboarding
onboardClient(clientEmail, clientName, calendarLink)
  .then(success => process.exit(success ? 0 : 1))
  .catch(error => {
    console.error('Fatal error:', error);
    process.exit(1);
  });
